packages/block-library/src/gallery/shared.js

```diff
@@ -23,25 +28,33 @@ export const pickRelevantMediaFiles = ( image, sizeSlug = 'large' ) => {
 	return imageProps;
 };
 
+function getGalleryBlockV2Enabled() {
+	// We want to fail early here, at least during beta testing phase, to ensure
+	// there aren't instances where undefined values cause false negatives.
+	if ( ! window.wp || typeof window.wp.galleryBlockV2Enabled !== 'boolean' ) {
+		throw 'window.wp.galleryBlockV2Enabled is not defined';
+	}
+	return window.wp.galleryBlockV2Enabled;
+}
+
 /**
  * The new gallery block format is not compatible with the use_BalanceTags option
  * in WP versions <= 5.8 https://core.trac.wordpress.org/ticket/54130. The
  * window.wp.galleryBlockV2Enabled flag is set in lib/compat.php. This method
  * can be removed when minimum supported WP version >=5.9.
  */
 export function isGalleryV2Enabled() {
+	// The logic for the native version is located in a different if statement
+	// due to the fact that the `process.env.IS_GUTENBERG_PLUGIN` constant
+	// should only be used as the condition in an if statement.
```

Comment:
Based on the code changes, my understanding is that we need to avoid referencing `process.env.IS_GUTENBERG_PLUGIN` on native platforms because the constant will be undefined. If that is accurate, the current wording of this comment does not clearly communicate that, in my opinion. WDYT about the following adjustments? 

```suggestion
	// The logic for the native version is located in a different if statement
	// due to the fact that `process.env.IS_GUTENBERG_PLUGIN` is
	// undefined on native platforms. 
```

Reply:
> Based on the code changes, my understanding is that we need to avoid referencing `process.env.IS_GUTENBERG_PLUGIN` on native platforms because the constant will be undefined.

It's true that `process.env.IS_GUTENBERG_PLUGIN` will always be undefined for the native version, but not sure if that's the actual reason for the split. In fact, we could have added the condition `Platform.isNative` to `if ( process.env.IS_GUTENBERG_PLUGIN )` statement. However, this throws the following lint error:

<img width="1378" alt="Screenshot 2022-06-14 at 11 07 15" src="https://user-images.githubusercontent.com/14905380/173540355-997505fc-313e-4ef6-a445-9161782cba3c.png">

Actually, the comment is based on the above error message, related to the [ESLint rule `@wordpress/is-gutenberg-plugin`](https://github.com/WordPress/gutenberg/blob/862ec612ed49fbb484a988954415771b3c828e0e/packages/eslint-plugin/rules/is-gutenberg-plugin.js#L60).

I think that in the end, the main reason for having the two `if` statements is basically because it's enforced by the lint rule. I agree that the comment might not be very accurate, so I'm wondering if we should expand it with the insights I added above, wdyt?

Reply:
Aha! Thank you for the additional context. I definitely did not comprehend that the code comment was related to a lint error. 

> I agree that the comment might not be very accurate, so I'm wondering if we should expand it with the insights I added above, wdyt?

Yes, adding further nuance to the comment would be helpful IMO. Maybe something like...

```suggestion
	// The logic for the native version is located in a different if statement
	// due to a lint rule that prohibits a single conditional combining
	// `process.env.IS_GUTENBERG_PLUGIN` with a native platform check.
```

On the other hand, if a lint error is thrown in the event someone attempts to combine the conditionals, then a code comment may not be necessary at all. I'll leave that decision to you. I am happy either way.

Reply:
> Aha! Thank you for the additional context. I definitely did not comprehend that the code comment was related to a lint error.

Yep, good point. I agree that would be a good idea to update the message.

> Yes, adding further nuance to the comment would be helpful IMO. Maybe something like...

I like the rephrasing, I'll apply it, thanks 🙇 !

> On the other hand, if a lint error is thrown in the event someone attempts to combine the conditionals, then a code comment may not be necessary at all. I'll leave that decision to you. I am happy either way.

Right, the comment is probably a bit redundant. But I think it would help to prevent contributors to be tempted to refactor it, since having the two `if` statements are odd in this part.